### DISCUSSION TITLE
ESP32 Board Compatibilty

### DISCUSSION THREAD

**Discussion by electric-m** [rednblkx/HomeKey-ESP32#24] [General]
Hey there, general question:

Are all ESP32 Boards compatible.
A have a spare dFrobot Beetle which is based on the ESP-WROOM-32.
Do we need to edit the code to implement different boards?
Any boards, that are especially known for working good?

michael

**Comment by rednblkx** [comment]
Hi there!

It should work with any board that has an ESP32 chip, not a ESP32-C2, S2 or others, so the ESP-WROOM-32 should be fine, i have an MH-ET D1 mini which uses ESP-WROOM-32.

**Reply by electric-m** [reply]
Thanks, I‘m gonna give it a try!

**Comment by 011V32** [comment]
Are there also some Ethernet PoE boards compatible?
For example the [ESP32 Ethernet Unit with PoE](https://shop.m5stack.com/products/esp32-ethernet-unit-with-poe) from M5Stack?
Or the [LiLiGo T-ETH-Lite ESP32](https://www.lilygo.cc/products/t-eth-lite?variant=43120880746677)?
Or the [Olimex ESP32-POE](https://www.olimex.com/Products/IoT/ESP32/ESP32-POE/open-source-hardware)?

**Reply by rednblkx** [reply]
I don't have any Etherner PoE board so i cannot attest whether it's compatible but as long as it's using an ESP32 chip, it should work, however, do note that Ethernet is not available as the base library HomeSpan relies on Wifi.

**Reply by 011V32** [reply]
Probably soon Ethernet support 🥳 
[https://github.com/HomeSpan/HomeSpan/issues/833#issuecomment-2048854468](https://github.com/HomeSpan/HomeSpan/issues/833#issuecomment-2048854468)

**Comment by aescolastico** [comment]
anyone try a 802.15.4 chip?

https://www.espressif.com/en/products/socs/esp32-c6
